### ISSUE TITLE
Feature request: Rust Cargo.toml dependencies auto completion

### ISSUE THREAD

**Issue by Vrajs16** [tamasfe/taplo#476]
Right now it is **not** possible to see which crates from crates.io are available underneath [dependencies] table in Cargo.toml.

Is there any plan to include this feature in the project's future roadmap? 

If so, I don't mind working on it.

**Comment by ia0** [comment]
There is no active work planned so you would indeed need to implement it.

I'm a bit worried of such feature increasing the risk of users falling for typosquat attacks. It's easy to auto-complete the wrong crate. See https://blog.phylum.io/rust-malware-staged-on-crates-io/ for examples. Ideally the auto-completion should sort the results by decreasing download or something like that.

(Note that this looks related to #472 and might share some code.)

**Comment by panekj** [comment]
I don't think it's our responsibility to provide such completions. Taplo should provide only what is given by schema, and any other additional information should be done via specialised extension/tooling (e.g.: https://github.com/serayuzgur/crates).

**Comment by Vrajs16** [comment]
Hey @ia0, do you mind helping me setup the project to atleast run the extension. Once I launch the extension I get a connection client error for some reason.

**Comment by ia0** [comment]
I'm not sure what you mean. You already can run the extension according to the bug report. Maybe you mean a custom development version? In that case (assuming Linux) you would need to change your `PATH` to also contain your `target/debug` or `target/release` directory, and restart your editor.

**Comment by Vrajs16** [comment]
Yes for a development. I am using a mac but should be simliar. Here are the steps I followed:
1. Cloned repo
2. Added directories to my `PATH` in my `~/.zshrc`
```zsh
export PATH=$PATH:$HOME/dev/repos/taplo/target/debug
export PATH=$PATH:$HOME/dev/repos/taplo/target/release
```
3. Ran `yarn install && yarn build` inside `taplo/editors/vscode`
4. Launched the extension from vscode launch.json file
5. Than I run into this error when I try to change the schema:

https://github.com/tamasfe/taplo/assets/89588024/a9b3439b-1041-4ae4-ac44-46125aa912cf

This has something to do with the lsp not being able to connect but I can't seem to figure it out. Am I missing something?
